File: sdks/ruby/lib/novabilling/credit_notes/types/create_credit_note_dto.rb
# frozen_string_literal: true

module Novabilling
  module CreditNotes
    module Types
      class CreateCreditNoteDto < Internal::Types::Model
        field :invoice_id, -> { String }, optional: false, nullable: false, api_name: "invoiceId"
        field :customer_id, -> { String }, optional: false, nullable: false, api_name: "customerId"
        field :amount, -> { Integer }, optional: false, nullable: false
        field :currency, -> { String }, optional: false, nullable: false
        field :reason, -> { Novabilling::CreditNotes::Types::CreateCreditNoteDtoReason }, optional: false, nullable: false
        field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
        field :status, -> { Novabilling::CreditNotes::Types::CreateCreditNoteDtoStatus }, optional: true, nullable: false
        field :created_at, -> { String }, optional: true, nullable: false, api_name: "createdAt"
      end
    end
  end
end

File: sdks/rust/src/api/types/create_credit_note_dto.rs
pub use crate::prelude::*;

/// Request type for API operation
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct CreateCreditNoteDto {
    /// Invoice ID to credit against
    #[serde(rename = "invoiceId")]
    pub invoice_id: String,
    /// Customer ID
    #[serde(rename = "customerId")]
    pub customer_id: String,
    /// Credit amount
    pub amount: f64,
    /// Currency
    pub currency: String,
    pub reason: CreateCreditNoteDtoReason,
    /// Additional metadata
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
    /// Override status for imports
    #[serde(skip_serializing_if = "Option::is_none")]
    pub status: Option<CreateCreditNoteDtoStatus>,
    /// Backdate createdAt (ISO 8601). For data imports.
    #[serde(rename = "createdAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
}
